src/VersionManager.cs

```diff
@@ -19,68 +20,162 @@ internal class VersionManager
-                    {
-                        Console.Error.WriteLine(Output.Red($"You must provide an invoice number in order to download Pro DarkRift releases."));
-                        return null;
-                    }
-
-                    uri += $"?invoice={invoiceNumber}";
+                    Console.Error.WriteLine(Output.Red($"You must provide an invoice number in order to download Pro DarkRift releases."));
+                    return false;
                 }
 
-                try
+                uri += $"?invoice={invoiceNumber}";
+            }
+
+            try
+            {
+                using (WebClient myWebClient = new WebClient())
                 {
-                    using (WebClient myWebClient = new WebClient())
-                    {
-                        myWebClient.DownloadFile(uri, stagingPath);
-                    }
+                    myWebClient.DownloadFile(uri, stagingPath);
                 }
-                catch (WebException e)
+            }
+            catch (WebException e)
+            {
+                Console.Error.WriteLine(Output.Red($"Could not download DarkRift {version} - {tier} (.NET {platform}):\n\t{e.Message}"));
+                return false;
+            }
+
+            Console.WriteLine($"Extracting package...");
+
+            Directory.CreateDirectory(fullPath);
+
+            ZipFile.ExtractToDirectory(stagingPath, fullPath, true);
+
+            Console.WriteLine(Output.Green($"Successfully downloaded package."));
+
+            return true;
+        }
+
+        public static bool IsVersionInstalled(Version version, ServerTier tier, ServerPlatform platform)
+        {
+            Version[] versions = GetVersionDirectories(tier, platform);
+            return Array.Exists(versions, (v) => v == version);
+        }
+
+        /// <summary>
+        /// Gets a list of version with specific tier and platform
+        /// </summary>
+        /// <param name="version">The version number required.</param>
+        /// <param name="pro">Whether the pro version should be used.</param>
+        /// <returns>The list of versions or an empty array</returns>
+        public static Version[] GetVersionDirectories(ServerTier tier, ServerPlatform platform)
+        {
+            try
+            {
+                string[] paths = Directory.GetDirectories(Path.Combine(USER_DR_DIR, "installed", $"{tier.ToString().ToLower()}", $"{platform.ToString().ToLower()}"));
+
+                Version[] versions = new Version[paths.Length];
+
+                // This removes the path and just leaves the version number
+                for (int i = 0; i < paths.Length; i++)
                 {
-                    Console.Error.WriteLine(Output.Red($"Could not download DarkRift {version} - {tier} (.NET {platform}):\n\t{e.Message}"));
-                    return null;
+                    FileInfo fi = new FileInfo(paths[i]);
+                    versions[i] = new Version(fi.Name);
```

Comment:
Can this not be done with [Paths.GetFileNameWithoutExtension](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getfilenamewithoutextension?view=netframework-4.8#System_IO_Path_GetFileNameWithoutExtension_System_String_). I could be very wrong on this one!